isis/tests/CSMCameraTests.cpp

```diff
@@ -0,0 +1,311 @@
+      testCube->write(csmStateBlob);
+      filename = testCube->fileName();
+      testCube->close();
+      testCube->open(filename);
+
+      testCam = testCube->camera();
+    }
+};
+
+TEST(CSMCameraTest, MockTest) {
+  MockRasterGM mockModel;
+  EXPECT_CALL(mockModel, getVersion())
+      .Times(1)
+      .WillOnce(::testing::Return(csm::Version(1, 2, 3)));
+  csm::Version mockVersion = mockModel.getVersion();
+  EXPECT_EQ(mockVersion.major(), 1);
+  EXPECT_EQ(mockVersion.minor(), 2);
+  EXPECT_EQ(mockVersion.revision(), 3);
+}
+
+TEST(CSMCameraTest, LoadMockTest) {
+  MockRasterGM mockModel;
+  EXPECT_CALL(mockModel, getVersion())
+      .Times(1)
+      .WillOnce(::testing::Return(csm::Version(1, 2, 3)));
+
+  // Use a universally unique identifier for thread safety
+  std::string mockModelName = QUuid().toString().toStdString();
+  MockCsmPlugin loadablePlugin;
+  loadablePlugin.registerModel(mockModelName, &mockModel);
+
+  csm::Model *returnedModel = csm::Plugin::findPlugin(MockCsmPlugin::PLUGIN_NAME)->constructModelFromState(mockModelName);
+  csm::Version mockVersion = returnedModel->getVersion();
+  EXPECT_EQ(mockVersion.major(), 1);
+  EXPECT_EQ(mockVersion.minor(), 2);
+  EXPECT_EQ(mockVersion.revision(), 3);
+}
+
+TEST_F(CSMCameraFixture, SetImage) {
+  csm::Ellipsoid wgs84;
+  EXPECT_CALL(mockModel, imageToRemoteImagingLocus(MatchImageCoord(csm::ImageCoord(4.5, 4.5)), ::testing::_, ::testing::_, ::testing::_))
+      .Times(1)
+      // looking straight down X-Axis
+      .WillOnce(::testing::Return(csm::EcefLocus(wgs84.getSemiMajorRadius() + 50000, 0, 0, -1, 0, 0)));
+
+  testCam->SetImage(5, 5);
+  EXPECT_EQ(testCam->UniversalLatitude(), 0.0);
+  EXPECT_EQ(testCam->UniversalLongitude(), 0.0);
+}
+
+TEST_F(CSMCameraFixture, SetGround) {
+  // Define some things to match/return
+  csm::Ellipsoid wgs84;
+  csm::ImageCoord imagePt(4.5, 4.5);
+  csm::EcefCoord groundPt(wgs84.getSemiMajorRadius(), 0, 0);
+  csm::EcefCoord observerPos(wgs84.getSemiMajorRadius() + 50000, 0, 0);
+
+  // Setup expected calls/returns
+  EXPECT_CALL(mockModel, groundToImage(MatchEcefCoord(groundPt), ::testing::_, ::testing::_, ::testing::_))
+      .Times(1)
+      .WillOnce(::testing::Return(imagePt));
+  EXPECT_CALL(mockModel, getSensorPosition(MatchImageCoord(imagePt)))
+      .Times(1)
+      .WillOnce(::testing::Return(observerPos));
+
+  testCam->SetGround(Latitude(0.0, Angle::Degrees), Longitude(0.0, Angle::Degrees));
+  EXPECT_EQ(testCam->Line(), 5.0);
+  EXPECT_EQ(testCam->Sample(), 5.0);
+}
+
+TEST_F(CSMCameraFixture, Resolution) {
+  // Setup to return the ground partials we want
+  // The pseudoinverse of:
+  // 1 2 3
+  // 4 5 6
+  //
+  // is
+  // -17  8
+  //  -2  2  *  1/18
+  //  13 -4
+  EXPECT_CALL(mockModel, computeGroundPartials)
+      .Times(6)
+      .WillRepeatedly(::testing::Return(std::vector<double>{1, 2, 3, 4, 5, 6}));
+
+  // We also have to set the mock up for setimage
```

Comment:
If we end up having to duplicate this code a lot to get a camera with a set image/ground point then we can move it into a new fixture.